onnx/test/schema_test.py

```diff
@@ -258,5 +259,104 @@ def test_init_with_default_value(self):
+            "op_domain": "",
+        },
+        {
+            "op_type": "CustomOp",
+            "op_version": 1,
+            "op_domain": "test",
+        },
+    ]
+)
+class TestOpSchemaRegister(unittest.TestCase):
+    op_type: str
+    op_version: int
+    op_domain: str
+
+    def setUp(self) -> None:
+        # Ensure the schema is unregistered
+        self.assertFalse(onnx.defs.has(self.op_type, self.op_domain))
+
+    def tearDown(self) -> None:
+        # Clean up the registered schema
+        with contextlib.suppress(onnx.defs.SchemaError):
+            onnx.defs.deregister_schema(self.op_type, self.op_version, self.op_domain)
+
+    def test_register_schema(self):
+        input = f"""
+            <
+                ir_version: 7,
+                opset_import: [
+                    "{self.op_domain}" : {self.op_version}
+                ]
+            >
+            agraph (float[N, 128] X, int32 Y) => (float[N] Z)
+            {{
+                Z = {self.op_domain}.{self.op_type}<attr1=[1,2]>(X, Y)
+            }}
+           """
+        model = onnx.parser.parse_model(input)
+        op_schema = defs.OpSchema(
+            self.op_type,
+            self.op_domain,
+            self.op_version,
+            inputs=[
+                defs.OpSchema.FormalParameter("input1", "T"),
+                defs.OpSchema.FormalParameter("input2", "int32"),
+            ],
+            outputs=[
+                defs.OpSchema.FormalParameter("output1", "T"),
+            ],
+            type_constraints=[("T", ["tensor(float)"], "")],
+            attributes=[
+                defs.OpSchema.Attribute(
+                    "attr1", defs.OpSchema.AttrType.INTS, "attr1 description"
+                )
+            ],
+        )
+        with self.assertRaises(onnx.checker.ValidationError):
+            onnx.checker.check_model(model, check_custom_domain=True)
+        onnx.defs.register_schema(op_schema)
+        self.assertTrue(onnx.defs.has(self.op_type, self.op_domain))
+        onnx.checker.check_model(model, check_custom_domain=True)
+
+        registered_op = onnx.defs.get_schema(
+            op_schema.name, op_schema.since_version, op_schema.domain
+        )
+        self.assertEqual(str(registered_op), str(op_schema))
+
+    def test_register_schema_raises_error_when_registering_a_schema_twice(self):
+        op_schema = defs.OpSchema(
+            self.op_type,
+            self.op_domain,
+            self.op_version,
+        )
+        onnx.defs.register_schema(op_schema)
+        with self.assertRaises(onnx.defs.SchemaError):
+            onnx.defs.register_schema(op_schema)
+
+    def test_deregister_schema(self):
+        op_schema = defs.OpSchema(
+            self.op_type,
+            self.op_domain,
+            self.op_version,
+        )
+        onnx.defs.register_schema(op_schema)
+        self.assertTrue(onnx.defs.has(op_schema.name, op_schema.domain))
+        onnx.defs.deregister_schema(
```

Comment:
If we get a schema from the registry, deregister it, then continue to access the schema, can we create a test to ensure the schema is still accessible?

Reply:
Good suggestion, found a bug.